### PR TITLE
Implement support for Bluetooth devices

### PR DESCRIPTION
**Pull request by caybro**
The Bluetooth backend uses QtBluetooth module, mention it in README.

The backend is capable of discovering and pairing of BT devices, 
and connecting to its various offered services.

### CODE REVIEW

**Review by zeenix — COMMENTED** [review]
It's good. Rationale not so important for such a small change but would prefer if you add a context prefix to the commit shortlog (e.g "bluetooth: ").

**Comment by zeenix** [comment]
my last comment was on "use QStringLiteral" patch, so annoying that github doesn't tell you the context. Although I think it applies to other commits too. Another thing that seems to apply to others is: Capitalization of the first character (not of the context prefix but the actual summary/shotlog).

File: src/lib/bluetooth/bluetoothbackend.cpp [hunk 1]
```diff
@@ -115,8 +115,6 @@ void BluetoothBackend::doStartDiscovery()
 void BluetoothBackend::stopDiscovery()
```
Comment (zeenix) [inline src/lib/bluetooth/bluetoothbackend.cpp]: Inline commenting so github knows which patch this is about.

As I said before "fix" is very vague verb that hardly ever summarizes the change. Please summarize the change itself and in the description, the benefits and/or rationale.

File: src/lib/bluetooth/bluetoothbackend.cpp [hunk 2]
```diff
@@ -36,6 +36,14 @@ BluetoothBackend::BluetoothBackend(QObject *parent)
 {
```
Comment (zeenix) [inline src/lib/bluetooth/bluetoothbackend.cpp:36]: It's good that BT connection works but please see my comments on shortlog rules on other patches.

Comment (caybro) [reply src/lib/bluetooth/bluetoothbackend.cpp:36]: Noted for the future, but remember all those sub-commits will get squashed when merged into master.
That said, I tried to follow the commit message rules for the main commit on top.

Comment (GordanM) [reply src/lib/bluetooth/bluetoothbackend.cpp:36]: If I am not mistaken we disable squash and merge commits on our repositories. This was decided on one of the compliance office meetings and documented by @DunderRoffe . Could you verify this please?

Comment (caybro) [reply src/lib/bluetooth/bluetoothbackend.cpp:36]: It's not disabled here, so I'll do it now, close this PR and send a new one ;)

**Comment by caybro** [comment]
Squash merge disabled, I squashed the commits locally and pushed here. Should be good now

**Comment by rpannek** [comment]
We disabled the squash merge so people would squash locally and not rely on the github squash magic from the github UI. It makes it much easier for the reviewers to see how the commits will end up in the master branch.

**Comment by caybro** [comment]
Yeah, understood @rpannek

**Comment by DunderRoffe** [comment]
You might already know this but anyway: one does not need to close this PR to squash locally. The PR will get updated whenever the branch that is was based on changes. So in the case of squashing it is enough to do it locally and then force push to the branch on your fork which this PR originates from.

**Review by rpannek — APPROVED** [review]

**Pull request merged**